### PR TITLE
LPS-180678 Add POSHI automation | Master

### PR DESCRIPTION
**Pull request by alessandromalves**
This is only for master
https://issues.liferay.com/browse/LPS-180678
cc.: @david-gutierrez-mesa @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
Please correct the commit message to use automation ticket number LPS-180678

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 1]
```diff
@@ -981,6 +981,41 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-178476. No alert should be displayed for a page with the XSS title visited by page tree."
+	@priority = 4
+	test ViewAlertsAreNotDisplayedForPagesWithXSSTitle {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:986]: Rename to AccessToPageTreeWherePageNameWithXSS
Move to PagesTreeWithXSS.testcase

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 2]
```diff
@@ -981,6 +981,41 @@ definition {
...
+		task ("Given a published content page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:987]: Given a user has a page with XSS in name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 3]
```diff
@@ -981,6 +981,41 @@ definition {
...
+				groupName = ${randomSiteName},
+				layoutName = "Test Page Name",
+				type = "content");
+
+			JSONLayout.publishLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Test Page Name");
+		}
+
+		task ("When user edit the content page title to a XSS title and goes to page tree") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:998]: When the user accesses to the Page Tree

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 4]
```diff
@@ -981,6 +981,41 @@ definition {
...
+				pageName = "Test Page Name");
+
+			PagesAdmin.editName(pageNameEdit = '''"xyz"><img src=x onerror=alert(1)>"''');
+
+			Navigator.openSiteURL(siteName = ${randomSiteName});
+
+			PagesFinder.goToPagesFinder();
+		}
+
+		task ("Then no alert should appear") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1012]: Then the user shouldn't see alert popup

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 5]
```diff
@@ -981,6 +981,41 @@ definition {
...
+		task ("When user edit the content page title to a XSS title and goes to page tree") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${randomSiteName});
+
+			PagesAdmin.gotoPageEllipsisMenuItem(
+				menuItem = "Configure",
+				pageName = "Test Page Name");
+
+			PagesAdmin.editName(pageNameEdit = '''"xyz"><img src=x onerror=alert(1)>"''');
+
+			Navigator.openSiteURL(siteName = ${randomSiteName});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1007]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 6]
```diff
@@ -981,6 +981,41 @@ definition {
...
+		}
+
+		task ("When user edit the content page title to a XSS title and goes to page tree") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${randomSiteName});
+
+			PagesAdmin.gotoPageEllipsisMenuItem(
+				menuItem = "Configure",
+				pageName = "Test Page Name");
+
+			PagesAdmin.editName(pageNameEdit = '''"xyz"><img src=x onerror=alert(1)>"''');
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1005]: Move to Given block and use more common pattern ```<script>alert(123);</script>```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 7]
```diff
@@ -981,6 +981,41 @@ definition {
...
+
+			Navigator.openSiteURL(siteName = ${randomSiteName});
+
+			PagesFinder.goToPagesFinder();
+		}
+
+		task ("Then no alert should appear") {
+			AssertElementNotPresent(locator1 = "Message#SUCCESS");
+
+			AssertElementNotPresent(locator1 = "Message#ERROR");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1015]: AssertAlertNotPresent();

**Review by Tim-Cao — COMMENTED** [review]
Hi @alessandromalves, would you mind refactoring according to comments? Thanks

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**